WordPress/Classes/ViewRelated/Blog/Subscribers/SubscribersView.swift

```diff
@@ -0,0 +1,131 @@
+import UIKit
+import SwiftUI
+import WordPressUI
+import WordPressKit
+
+final class SubscribersViewController: UIHostingController<AnyView> {
+    private let viewModel: SubscribersViewModel
+
+    init(blog: Blog) {
+        self.viewModel = SubscribersViewModel(blog: blog)
+        super.init(rootView: AnyView(SubscribersView(viewModel: viewModel)))
+        self.title = Strings.title
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+private struct SubscribersView: View {
+    @ObservedObject var viewModel: SubscribersViewModel
+
+    var body: some View {
+        Group {
+            if !viewModel.searchText.isEmpty {
+                SubscribersSearchView(viewModel: viewModel)
+            } else {
+                SubscribersListView(viewModel: viewModel)
+            }
+        }
+        .searchable(text: $viewModel.searchText)
+    }
+}
+
+private struct SubscribersListView: View {
+    @ObservedObject var viewModel: SubscribersViewModel
+
+    var body: some View {
+        List {
+            if let response = viewModel.response {
+                SubscribersPaginatedForEach(response: response)
+            }
+        }
+        .listStyle(.plain)
+        .overlay {
+            if let response = viewModel.response {
+                if response.isEmpty {
+                    EmptyStateView(Strings.empty, systemImage: "envelope")
+                }
+            } else if viewModel.isLoading {
+                ProgressView()
+            } else if let error = viewModel.error {
+                EmptyStateView.failure(error: error) {
+                    Task { await viewModel.refresh() }
+                }
+            }
+        }
+        .task(id: viewModel.parameters) {
+            await viewModel.refresh()
+        }
+        .refreshable {
+            await viewModel.refresh()
+        }
+        .toolbar {
+            ToolbarItem(placement: .topBarTrailing) {
+                SubscribersMenu(viewModel: viewModel)
+            }
+        }
+    }
+}
+
+private struct SubscribersSearchView: View {
+    @ObservedObject var viewModel: SubscribersViewModel
+
+    @State private var response: SubscribersPaginatedResponse?
+    @State private var error: Error?
+
+    var body: some View {
+        List {
+            if let response {
+                SubscribersPaginatedForEach(response: response)
+            } else if error == nil {
+                ListFooterView(.loading)
+            }
+        }
+        .listStyle(.plain)
+        .overlay {
+            if let response, response.isEmpty {
+                EmptyStateView.search()
+            } else if let error {
+                EmptyStateView.failure(error: error)
+            }
+        }
+        .task(id: viewModel.searchText) {
+            error = nil
+            do {
+                try await Task.sleep(for: .milliseconds(500))
```

Comment:
Why `sleep` here?

Reply:
I challenge myself to only use Swift Concurrency and avoid Combine, and this seems like how you typically implement debounce. It effectively performs the request when input changes only after idling for 500ms.